### PR TITLE
BaSP-TG-54: fix timesheets pages crash

### PR DESCRIPTION
**Pull request by visojulian**
Timesheets page crashes if any field in timsheet rows list is undefinded
Shloud skip the row and render anyway

---
Added condition to the timsheets table row render

### CODE REVIEW

**Review by baudraccoverona — DISMISSED** [review]

File: src/Components/Admins/AdminList/AdminList.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import styles from '../admins.module.css';
 import AdminItem from './AdminItem';
-import Delete from '../../../Assets/trash.png';
+import Delete from '../../../assets/trash.png';
```
Comment (Licamus) [inline src/Components/Admins/AdminList/AdminList.jsx:4]: Assets with mayus.

File: src/Components/Tasks/index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { useEffect, useState } from 'react';
 import styles from './tasks.module.css';
 import Task from './Task/index';
 import Modal from './Modal/index';
-import Logo from '../../Assets/trash.png';
+import Logo from '../../assets/trash.png';
```
Comment (Licamus) [inline src/Components/Tasks/index.js:5]: Assets with mayus

File: src/Components/TimeSheets/TimeSheet/index.js [hunk 3]
```diff
@@ -1,71 +1,45 @@
-import Trash from '../../../Assets/trash.png';
+import Trash from '../../../assets/trash.png';
```
Comment (Licamus) [inline src/Components/TimeSheets/TimeSheet/index.js:1]: Assets with Mayus

File: src/Components/Admins/AdminList/AdminList.jsx (hunk 1 above)
Comment (lmalvasia) [reply src/Components/Admins/AdminList/AdminList.jsx:4]: I think that it's okey the name with lowercase.

File: src/Components/TimeSheets/TimeSheet/index.js [hunk 4]
```diff
@@ -1,71 +1,45 @@
...
 import styles from './time.module.css';
 
 const TimeSheet = ({ timeSheet, setShowModal, setTimeSheetId }) => {
-  return (
-    <tr style={{ display: 'flex', justifyContent: 'center' }}>
-      <td
+  if (
```
Comment (lmalvasia) [inline src/Components/TimeSheets/TimeSheet/index.js:5]: The problem happens only when the task, employee or project entity is deleted, so we should show the table with all the available information. You have to add a couple of conditions in another side (check next comments 😂 )

File: src/Components/TimeSheets/TimeSheet/index.js [hunk 5]
```diff
@@ -1,71 +1,45 @@
...
-            setTimeSheetId(timeSheet._id);
-          }}
-        />
-      </td>
-    </tr>
-  );
+        <td style={{ flexBasis: '25%' }}>{timeSheet.description}</td>
+        <td style={{ flexBasis: '15%' }}>{timeSheet.date}</td>
+        <td style={{ flexBasis: '10%' }}>{timeSheet.hours}</td>
+        <td style={{ flexBasis: '20%' }}>{timeSheet.task.description}</td>
```
Comment (lmalvasia) [inline src/Components/TimeSheets/TimeSheet/index.js:23]:
```suggestion
        <td style={{ flexBasis: '20%' }}>{timeSheet.task ? timeSheet.task.description : 'Task does not exist'}</td>
```

File: src/Components/TimeSheets/TimeSheet/index.js [hunk 6]
```diff
@@ -1,71 +1,45 @@
...
+        <td style={{ flexBasis: '20%' }}>
+          {`${timeSheet.employee.name} ${timeSheet.employee.lastName}`}
```
Comment (lmalvasia) [inline src/Components/TimeSheets/TimeSheet/index.js:25]:
```suggestion
          {timeSheet.employee ? `${timeSheet.employee.name} ${timeSheet.employee.lastName}` : 'Employee does not exist'}
```

File: src/Components/TimeSheets/TimeSheet/index.js [hunk 7]
```diff
@@ -1,71 +1,45 @@
...
+        </td>
+        <td style={{ flexBasis: '10%' }}>{timeSheet.project.name}</td>
```
Comment (lmalvasia) [inline src/Components/TimeSheets/TimeSheet/index.js:27]:
```suggestion
        <td style={{ flexBasis: '10%' }}>{timeSheet.project ? timeSheet.project.name : 'Project does not exist'}</td>
```

File: src/Components/TimeSheets/TimeSheet/index.js (hunk 4 above)
Comment (visojulian) [reply src/Components/TimeSheets/TimeSheet/index.js]: Done!

**Review by lmalvasia — DISMISSED** [review]

**Review by Romimonde — DISMISSED** [review]
Thank you! This will be helpful to keep going and continue working. Although this doesn´t fix the last week bugs. I can still create a timeSheet with inexistent data

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**